lib/app/ui/widgets/shared/bottom_nav.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:cook_snap/app/constants/routes.dart';
+import 'package:cook_snap/app/ui/widgets/shared/bottom_nav_item.dart';
+import 'package:flutter/material.dart';
+
+class BottomNavBar extends StatelessWidget {
+  final int currentIndex;
+  final void Function(int) onIndexChanged;
+
+  const BottomNavBar({
+    super.key,
+    required this.currentIndex,
+    required this.onIndexChanged,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return BottomAppBar(
+      color: Colors.white,
+      height: 60.0,
+      padding: const EdgeInsets.symmetric(horizontal: 0.0, vertical: 0.0),
+      child: Container(
+        color: Colors.white,
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: AppRoutes.bottomNavBarRoutes.map((route) {
+            return Expanded(
+              child: BottomNavBarItem(
+                icon: route['icon'],
+                label: route['label'],
+                isActive: currentIndex == route['id'],
+                isCallToAction: route['id'] == 2,
+                onPressed: () {
+                  if (route['id'] == 2) {
+                    return;
+                  }
+                  onIndexChanged(route['id'] as int);
+                },
+              ),
+            );
```

Comment:
Nice work on the bottom nav bar, it's overflowing by 11px on my phone, not sure how you'd address that, I'd suggest making the labels shorter, I think the same issue is making the `Scan` label misplaced regarding the scan `fab`, look:

<img width="414" alt="image" src="https://github.com/didierganthier/cook-snap/assets/45739963/3f4009fc-8c0f-419f-8218-8e7cde8ea7da">

EDIT: I fixed it by wrapping the `BottomNavBarItem` with an `Expanded` widget, this caused an overflow on the bottom which I fixed by changing the `Notifications` label to `Alerts` thus making it shorter:

![image](https://github.com/didierganthier/cook-snap/assets/45739963/33b99033-df59-4da2-80e2-569a6fb032f4)